test/base/src/androidTest/kotlin/io/realm/test/shared/RealmListTests.kt

```diff
@@ -819,6 +858,31 @@ internal abstract class ManagedListTester<T>(
         }
     }
 
+    override fun reassign() {
+        val dataSet = typeSafetyManager.getInitialDataSet()
+        val reassignedDataSet = listOf(dataSet[1])
+
+        val assertions = { list: RealmList<T> ->
+            assertEquals(1, list.size)
+            // We cannot assert equality on RealmObject lists as the object isn't equals to the
+            // unmanaged object from before the assignment
+            if (list[0] !is RealmObject) {
```

Comment:
Would it be possible to check some property on the object instead, just as a smoke test? Right now RealmObjects are kinda slipping through this test.

Reply:
We already assert that the list is actually updated since the initial lists have two elements (guaranteed by accessing index 1) and the resulting list only having one (`list.size == 1` from above assert). But I can probably find something to assert.